File: sol/solution/0001-0100/0001. Two Sum/Solution.kt
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        val m = mutableMapOf<Int, Int>()
        nums.forEachIndexed { i, x ->
            val j = m.get(target - x)
            if (j != null) {
                return intArrayOf(j, i)
            }
            m[x] = i
        }
        return intArrayOf()
    }
}

File: sol/solution/0001-0100/0001. Two Sum/Solution.php
class Solution {
    /**
     * @param Integer[] $nums
     * @param Integer $target
     * @return Integer[]
     */
    function twoSum($nums, $target) {
        $d = [];
        foreach ($nums as $i => $x) {
            if (isset($d[$target - $x])) {
                return [$d[$target - $x], $i];
            }
            $d[$x] = $i;
        }
    }
}